popup.js

```diff
@@ -1,30 +1,44 @@
-var months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
-var days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]
+/**
+ * Over-engineered top-level closure to avoid
+ * anything here from leaking globally.
+ * @param {HTMLElement} timeEl
+ * @param {HTMLElement} dateEl
+ */
+;((timeEl, dateEl) => {
+  ;('use strict')
 
-function clockWrite() {
-	//Set all the time variables
-	var d = new Date(),
-	h = d.getUTCHours(),
-	m = d.getUTCMinutes(),
-	s = d.getUTCSeconds();
-	
-	//Set all the date variables
-	day	= d.getUTCDay(),
-	date = d.getUTCDate(),
-	month = d.getUTCMonth(),
-	year = d.getUTCFullYear();
-	
-	h = (h < 10 ? '0' : '') + h;
-	m = (m < 10 ? '0' : '') + m;
-	s = (s < 10 ? '0' : '') + s;
-	
-	utctime = (h + ':' + m + ':' + s);
-	utcdate = (days[day] + ' ' + months[month] + ' ' + date + ', ' + year);
-	
-	document.getElementById("time").firstChild.nodeValue = utctime;
-	document.getElementById("date").firstChild.nodeValue = utcdate;
-}
+  /** @type {Intl.DateTimeFormatOptions} */
```

Comment:
```ts
  /** @type {Intl.DateTimeFormatOptions} */
```
This is a type hint embedded in a comment using the JSDoc syntax. The TypeScript team has taken great pains to ensure compatibility with JSDoc's syntax, and the result is that you can add types without needing to know TypeScript's syntax.  As these type hints are inside of comments, you don't need to run a special build step to turn TS into JS.  So your code editor gets extra info to help you with autocomplete, but you don't wind up having to go all the way down the TS rabbit hole. 😉